tests/contrib/pytest/test_pytest_atr.py

```diff
@@ -0,0 +1,244 @@
+from ddtrace.contrib.pytest._utils import _pytest_version_supports_atr
+from ddtrace.internal.ci_visibility._api_client import TestVisibilityAPISettings
+from tests.ci_visibility.util import _get_default_civisibility_ddconfig
+from tests.contrib.pytest.test_pytest import PytestTestCaseBase
+from tests.contrib.pytest.test_pytest import _get_spans_from_list
+
+
+pytestmark = pytest.mark.skipif(
+    not (_USE_PLUGIN_V2 and _pytest_version_supports_atr()),
+    reason="Auto Test Retries requires v2 of the plugin and pytest >=7.0",
+)
+
+_TEST_PASS_CONTENT = """
+def test_func_pass():
+    assert True
+"""
+
+_TEST_FAIL_CONTENT = """
+import pytest
+
+def test_func_fail():
+    assert False
+
+_test_func_retries_skip_count = 0
+def test_func_retries_skip():
+    global _test_func_retries_skip_count
+    _test_func_retries_skip_count += 1
+    if _test_func_retries_skip_count > 1:
+        pytest.skip()
+    assert False
+
+"""
+
+_TEST_PASS_ON_RETRIES_CONTENT = """
+_test_func_passes_4th_retry_count = 0
+def test_func_passes_4th_retry():
+    global _test_func_passes_4th_retry_count
+    _test_func_passes_4th_retry_count += 1
+    assert _test_func_passes_4th_retry_count == 5
+
+_test_func_passes_1st_retry_count = 0
+def test_func_passes_1st_retry():
+    global _test_func_passes_1st_retry_count
+    _test_func_passes_1st_retry_count += 1
+    assert _test_func_passes_1st_retry_count == 2
+"""
+
+_TEST_ERRORS_CONTENT = """
+import pytest
+
+@pytest.fixture
+def fixture_fails_setup():
+    assert False
+
+def test_func_fails_setup(fixture_fails_setup):
+    assert True
+
+@pytest.fixture
+def fixture_fails_teardown():
+    yield
+    assert False
+
+def test_func_fails_teardown(fixture_fails_teardown):
+    assert True
+"""
+
+_TEST_SKIP_CONTENT = """
+import pytest
+
+@pytest.mark.skip
+def test_func_skip_mark():
+    assert True
+
+def test_func_skip_inside():
+    pytest.skip()
+"""
+
+
+class PytestATRTestCase(PytestTestCaseBase):
+    @pytest.fixture(autouse=True, scope="function")
+    def set_up_atr(self):
+        with mock.patch(
+            "ddtrace.internal.ci_visibility.recorder.CIVisibility._check_enabled_features",
+            return_value=TestVisibilityAPISettings(flaky_test_retries_enabled=True),
+        ):
+            yield
+            from ddtrace.internal.ci_visibility.recorder import CIVisibility
+
+            if CIVisibility.enabled:
+                CIVisibility.disable()
+
+    def test_pytest_atr_no_ddtrace_does_not_retry(self):
+        self.testdir.makepyfile(test_pass=_TEST_PASS_CONTENT)
+        self.testdir.makepyfile(test_fail=_TEST_FAIL_CONTENT)
+        self.testdir.makepyfile(test_errors=_TEST_ERRORS_CONTENT)
+        self.testdir.makepyfile(test_pass_on_retries=_TEST_PASS_ON_RETRIES_CONTENT)
+        self.testdir.makepyfile(test_skip=_TEST_SKIP_CONTENT)
+        rec = self.inline_run()
+        rec.assertoutcome(passed=2, failed=6, skipped=2)
+        assert len(self.pop_spans()) == 0
```

Comment:
It's not clear to me how the span numbers correlate to the tests run, can you clarify?

Reply:
There's a pair of implicit assumptions here:
- if we did not create spans (eg: the `ddtrace` plugin was not active), then we didn't retry tests
- if we did not retry tests, the outcome should be correct (eg: we run 10 tests total, and the ones that failed did not pass thanks to retries)

TBH, it doesn't add that much on top of the [`test_disabled`](https://github.com/DataDog/dd-trace-py/pull/11241/files/b444a492eb519a3597f99e39426056af0c09ff3e#diff-48a91ef11a9dbcd767fd3050fa39ca48a9bc8454d47d71b2afc35ebc630686f7R186) test it's based off of.